src/twisted/conch/ssh/factory.py

```diff
@@ -56,7 +56,7 @@ def buildProtocol(self, addr):
         @return: The built transport.
         """
         t = protocol.Factory.buildProtocol(self, addr)
-        t.supportedPublicKeys = self.privateKeys.keys()
+        t.supportedPublicKeys = list(self.privateKeys.keys())
```

Comment:
(See my [discussion comment](https://github.com/twisted/twisted/pull/1553#issuecomment-805348538) and [proposed change](https://github.com/nielsavonds/twisted/pull/1) )

Reply:
No response from the requestor in two months, so I went ahead and added my proposed fix to this PR.

Reply:
Hi @wiml thanks for the update. Let me know if you need a review on this.

Reply:
Also, let me know if you need help with writing an automated tests for this.